sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala

```diff
@@ -187,10 +187,11 @@ case class TargetSize(override val targetSizeBytes: Long) extends CoalesceSizeGo
  * for a key, the batch may still run into limits on set by Spark or cudf. It should be noted
  * that it is required that a node in the Spark plan that requires this should also require
  * an input ordering that satisfies this ordering as well.
- * @param order the keys that should be used for batching.
+ * @param gpuOrder the GPU keys that should be used for batching.
+ * @param cpuOrder the CPU keys that should be used for batching.
  */
-case class BatchedByKey(order: Seq[SortOrder]) extends CoalesceGoal {
-  override def children: Seq[Expression] = order
+case class BatchedByKey(gpuOrder: Seq[SortOrder], cpuOrder: Seq[SortOrder]) extends CoalesceGoal {
+  override def children: Seq[Expression] = gpuOrder ++ cpuOrder
```

Comment:
this looks odd, why is it both gpu and cpu order?  perhaps a comment

Reply:
I need to look into this further.  I originally left it as just the GPU SortOrder, but if I do then it leads to an `IndexOutOfBoundsException` during WindowFunctionSuite.  It's as if something sees the case class arguments, expects two children from the `children` result, but only finds one.

Reply:
So this ended up being messy because it's an expression that has two sequences in it now.  I ended up modeling it after [WindowSpecDefinition](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala#L44-L56) in Spark where they needed to have two separate sequences as arguments to an expression.  In order to override `withNewChildren` without shimming the class, I needed to extend our base shim tree node classes to allow a common override across Spark versions.  cc: @gerashegalov who will probably be interested in checking what I did there.

Reply:
Now that this is passing the CPU expressions as an separate parameter list, we don't need to override `withNewChildren` and thus don't need the changes to the shims and can return just the GPU expressions as children as it did before.